internal/service/ivs/channel.go

```diff
@@ -0,0 +1,272 @@
+	"github.com/hashicorp/terraform-provider-aws/names"
+)
+
+func ResourceChannel() *schema.Resource {
+	return &schema.Resource{
+		CreateWithoutTimeout: resourceChannelCreate,
+		ReadWithoutTimeout:   resourceChannelRead,
+		UpdateWithoutTimeout: resourceChannelUpdate,
+		DeleteWithoutTimeout: resourceChannelDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(5 * time.Minute),
+			Delete: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"authorized": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Computed: true,
+			},
+			"ingest_endpoint": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"latency_mode": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validation.StringInSlice(ivs.ChannelLatencyMode_Values(), false),
+			},
+			"name": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validation.StringMatch(regexp.MustCompile(`^[a-zA-Z0-9-_]{0,128}$`), "must contain only alphanumeric characters, hyphen, or underscore and at most 128 characters"),
+			},
+			"playback_url": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"recording_configuration_arn": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: verify.ValidARN,
+			},
+			"tags":     tftags.TagsSchema(),
+			"tags_all": tftags.TagsSchemaComputed(),
+			"type": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validation.StringInSlice(ivs.ChannelType_Values(), false),
+			},
+		},
+
+		CustomizeDiff: verify.SetTagsDiff,
+	}
+}
+
+const (
+	ResNameChannel = "Channel"
+)
+
+func resourceChannelCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*conns.AWSClient).IVSConn
+
+	in := &ivs.CreateChannelInput{}
+
+	if v, ok := d.GetOk("authorized"); ok {
+		in.Authorized = aws.Bool(v.(bool))
+	}
+
+	if v, ok := d.GetOk("latency_mode"); ok {
+		in.LatencyMode = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("name"); ok {
+		in.Name = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("recording_configuration_arn"); ok {
+		in.RecordingConfigurationArn = aws.String(v.(string))
+	}
+
+	defaultTagsConfig := meta.(*conns.AWSClient).DefaultTagsConfig
+	tags := defaultTagsConfig.MergeTags(tftags.New(d.Get("tags").(map[string]interface{})))
+
+	if len(tags) > 0 {
+		in.Tags = Tags(tags.IgnoreAWS())
+	}
+
+	if v, ok := d.GetOk("type"); ok {
+		in.Type = aws.String(v.(string))
+	}
+
+	out, err := conn.CreateChannelWithContext(ctx, in)
```

Comment:
The [`CreateChannelOutput`](https://pkg.go.dev/github.com/aws/aws-sdk-go@v1.44.133/service/ivs#CreateChannelOutput) contains both a `Channel` and `StreamKey` fields, however the [`GetChannelOutput`](https://pkg.go.dev/github.com/aws/aws-sdk-go@v1.44.133/service/ivs#GetChannelOutput) only contains the `Channel`. This has repercussions in the "return read after create" paradigm. I am unsure if the `aws_ivs_channel` schema should also have a `stream_key` computed field which could be populated by calling `ListStreamKeys` and `GetStreamKey`. 

The fourth and final PR of the set will introduce an `aws_ivs_stream_key` data source which retrieves this information via the above API calls, so a Terraform file could look something like:

```terraform
resource "aws_ivs_channel" "channel" {
  name = "channel-1"
}

data "aws_ivs_stream_key" "stream_key" {
  channel_arn = aws_ivs_channel.channel.arn
}
```

Just not sure if the key information should also be part of the channel, considering odds are that a user would generally want the stream key information on channel creation as well.

What are your thoughts?

Reply:
If `StreamKey` isn't used elsewhere then I would suggest leaving it out.
Having attributes set just in `Create` prevents resource import working correctly.